src/main/java/gregtech/loaders/recipe/RecyclingRecipes.java

```diff
@@ -483,4 +496,22 @@ private static void shrinkStacks(List<Tuple<ItemStack, MaterialStack>> list, Ite
     private static boolean isAshMaterial(MaterialStack ms) {
         return ms.material == Materials.Ash || ms.material == Materials.DarkAsh || ms.material == Materials.Carbon;
     }
+
+    /**
+     * Performs various NBT matching on the provided input and adds the result to the provided RecipeBuilder
+     *
+     * @param input The input itemStack
+     * @param builder The RecipeBuilder to add the NBT condition to
+     */
+    private static void cleanInputNBT(ItemStack input, RecipeBuilder<?> builder) {
+
+        // Ignore String tag from naming machines
+        MetaTileEntity mte = GTUtility.getMetaTileEntity(input);
+        if (mte != null) {
+            builder.clearInputs();
+            // Don't use ANY to avoid issues with Drums, Super Chests, and other MTEs that hold an inventory
+            NBTCondition condition = NBTCondition.create(NBTTagType.STRING, GregtechDataCodes.CUSTOM_NAME, "");
```

Comment:
This NBTCondition can be saved as a constant to reduce memory allocation/GC pressure during recipe registration. This method will be called a lot of times.